dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -895,18 +895,42 @@ Spdp::match_unauthenticated(const DCPS::RepoId& guid, DiscoveredParticipantIter&
 void
 Spdp::handle_auth_request(const DDS::Security::ParticipantStatelessMessage& msg)
 {
+  const RepoId guid = make_id(msg.message_identity.source_guid, DCPS::ENTITYID_PARTICIPANT);
+
+  if (DCPS::security_debug.auth_debug) {
+    ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {auth_debug} DEBUG: Spdp::handle_auth_request() - ")
+               ACE_TEXT("%C -> %C local %C\n"),
+               DCPS::LogGuid(guid).c_str(),
+               DCPS::LogGuid(msg.destination_participant_guid).c_str(),
+               DCPS::LogGuid(guid_).c_str()));
+  }
+
   // If this message wasn't intended for us, ignore handshake message
-  if (msg.destination_participant_guid != guid_ || msg.message_data.length() == 0) {
+  if (msg.destination_participant_guid != guid_) {
+    if (DCPS::security_debug.auth_debug) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {auth_debug} DEBUG: Spdp::handle_auth_request() - ")
+                 ACE_TEXT("Dropped not recipient\n")));
+    }
     return;
   }
 
-  const RepoId guid = make_id(msg.message_identity.source_guid, DCPS::ENTITYID_PARTICIPANT);
+  if (msg.message_data.length() == 0) {
+    if (DCPS::security_debug.auth_debug) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {auth_debug} DEBUG: Spdp::handle_auth_request() - ")
+                 ACE_TEXT("Dropped no data\n")));
+    }
+    return;
+  }
 
   ACE_GUARD(ACE_Thread_Mutex, g, lock_);
 
   if (sedp_->ignoring(guid)) {
     // Ignore, this is our domain participant or one that the user has
     // asked us to ignore.
+    if (DCPS::security_debug.auth_debug) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {auth_debug} DEBUG: Spdp::handle_auth_request() - ")
```

Comment:
Log the guid?

Reply:
Guids are logged upon entry into the function.